src/main/groovy/com/lesfurets/jenkins/unit/BasePipelineTest.groovy

```diff
@@ -100,24 +105,30 @@ abstract class BasePipelineTest {
         helper.registerAllowedMethod("input", [String.class], null)
         helper.registerAllowedMethod("junit", [String.class], null)
         helper.registerAllowedMethod("logRotator", [Map.class], null)
+        helper.registerAllowedMethod('mail', [Map.class], null)
         helper.registerAllowedMethod("node", [Closure.class], null)
         helper.registerAllowedMethod("node", [String.class, Closure.class], null)
         helper.registerAllowedMethod("pipelineTriggers", [List.class], null)
         helper.registerAllowedMethod("properties", [List.class], null)
         helper.registerAllowedMethod("readFile", [String.class], null)
-        helper.registerAllowedMethod('retry', [Integer.class, Closure.class], null)
+        helper.registerAllowedMethod('retry', [Integer.class, Closure.class]) { Integer count, Closure body ->
+            body()
```

Comment:
other mock uses [helper.callClosure](https://github.com/jenkinsci/JenkinsPipelineUnit/blob/85861a35547e4f6aea018790745ca6d52036b907/src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy#L499)
I'm not sure why, cannot see any unit test covering it, but I prefer to keep re-use same approach.
```
body.delegate = delegate
helper.callClosure(body)
```

Reply:
Great point, I've pushed 2 fixups to address this for `dir` and `retry`. Please let me know when you are OK with the state of this PR and I'll squash them before merge.

As for the use-case, here's an example: https://github.com/AbletonDevTools/postgres-pipeline-utils/blob/be944d643a0528a605c959b152426dd227bc10a5/src/com/ableton/PostgresDocker.groovy#L107-L114

We have several unit tests for the above code which assert that the stuff inside of `retry` ran correctly.

Reply:
thank you